### PR TITLE
apps/example/ta_tc/messaging/utc : Fix coverity issues

### PR DESCRIPTION
**Pull request by hasw7569**
Fix coverity issues that uninitialized pointer read in utc_messaging_send.c and double free in utc_messaging_recv.c

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [59d5261a39958429c1231aeb064779135e47f2c8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [59d5261a39958429c1231aeb064779135e47f2c8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ebd746605629d2a7d95b299a8f4ebba8b74d773f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ebd746605629d2a7d95b299a8f4ebba8b74d773f] - Code Rule Check (C++) OK.

File: apps/examples/testcase/ta_tc/messaging/utc/utc_messaging_recv.c [hunk 1]
```diff
@@ -76,6 +76,8 @@ static void block_sender(int argc, FAR char *argv[])
 
 	data.msglen = strlen(TC_BLOCK_MSG) + 1;
 	data.priority = 100;
+	data.msg = NULL;
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/messaging/utc/utc_messaging_recv.c]: Why is this necessary?

File: apps/examples/testcase/ta_tc/messaging/utc/utc_messaging_recv.c [hunk 2]
```diff
@@ -108,6 +110,8 @@ static void block_receiver(int argc, FAR char *argv[])
 	}
 
 	recv_buf.buflen = strlen(TC_BLOCK_MSG) + 1;
+	recv_buf.buf = NULL;
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/messaging/utc/utc_messaging_recv.c]: same as previous

File: apps/examples/testcase/ta_tc/messaging/utc/utc_messaging_recv.c [hunk 3]
```diff
@@ -152,6 +156,7 @@ static void nonblock_callback(msg_reply_type_t msg_type, msg_recv_buf_t *recv_da
 {
 	int ret;
 	msg_send_data_t reply;
+	reply.msg = NULL;
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/messaging/utc/utc_messaging_recv.c]: If line 161 is false, line 181 tries to free null. I think you should move line 181 into if case.

File: apps/examples/testcase/ta_tc/messaging/utc/utc_messaging_recv.c (hunk 1 above)
Comment (hasw7569) [reply apps/examples/testcase/ta_tc/messaging/utc/utc_messaging_recv.c]: It is not necessary.
I will remove this line.

File: apps/examples/testcase/ta_tc/messaging/utc/utc_messaging_recv.c (hunk 3 above)
Comment (hasw7569) [reply apps/examples/testcase/ta_tc/messaging/utc/utc_messaging_recv.c]: I agree with your opinion.
I will move line 181 into end of if case

**Comment by seinfra** [comment]
Target : [c1d5a9d0b1554b20f05b9bff5cbc25a3523383a0] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c1d5a9d0b1554b20f05b9bff5cbc25a3523383a0] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cdf88c5054e4e9439edefa9a7e7bdbd849f23a2d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cdf88c5054e4e9439edefa9a7e7bdbd849f23a2d] - Code Rule Check OK.
